presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/ThriftHiveRecordCursorProvider.java

```diff
@@ -0,0 +1,114 @@
+import com.facebook.presto.spi.type.TypeManager;
+import com.google.common.collect.ImmutableSet;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapred.RecordReader;
+import org.joda.time.DateTimeZone;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+
+import static com.facebook.presto.hive.HiveUtil.createRecordReader;
+import static com.facebook.presto.hive.HiveUtil.getDeserializerClassName;
+import static com.facebook.presto.hive.HiveUtil.getSerializationClassName;
+import static java.util.Objects.requireNonNull;
+
+public class ThriftHiveRecordCursorProvider
+        implements HiveRecordCursorProvider
+{
+    private static final String LAZY_BINARY_SERDE = "org.apache.hadoop.hive.serde2.lazybinary.LazyBinarySerDe";
+    private static final String THRIFT_GENERIC_ROW = ThriftGenericRow.class.getName();
+    private static final Set<String> THRIFT_SERDE_CLASS_NAMES = ImmutableSet.<String>builder()
+            .add("com.facebook.presto.twitter.hive.thrift.ThriftGeneralSerDe")
+            .add("com.facebook.presto.twitter.hive.thrift.ThriftGeneralDeserializer")
+            .add(LAZY_BINARY_SERDE)
+            .build();
+    private final HdfsEnvironment hdfsEnvironment;
+    private final ThriftFieldIdResolverFactory thriftFieldIdResolverFactory;
+
+    @Inject
+    public ThriftHiveRecordCursorProvider(HdfsEnvironment hdfsEnvironment, ThriftFieldIdResolverFactory thriftFieldIdResolverFactory)
+    {
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.thriftFieldIdResolverFactory = requireNonNull(thriftFieldIdResolverFactory, "thriftFieldIdResolverFactory is null");
+    }
+
+    @Override
+    public Optional<RecordCursor> createRecordCursor(
+            String clientId,
+            Configuration configuration,
+            ConnectorSession session,
+            Path path,
+            long start,
+            long length,
+            Properties schema,
+            List<HiveColumnHandle> columns,
+            TupleDomain<HiveColumnHandle> effectivePredicate,
+            DateTimeZone hiveStorageTimeZone,
+            TypeManager typeManager)
+    {
+        if (!THRIFT_SERDE_CLASS_NAMES.contains(getDeserializerClassName(schema))) {
+            return Optional.empty();
+        }
+
+        // We only allow the table which specified its serialization class is compatible to
+        // our thrift general row, if the SerDe is LazyBinarySerDe.
+        if (LAZY_BINARY_SERDE.equals(getDeserializerClassName(schema)) && !THRIFT_GENERIC_ROW.equals(getSerializationClassName(schema))) {
+            return Optional.empty();
+        }
+
+        setPropertyIfUnset(schema, "elephantbird.mapred.input.bad.record.check.only.in.close", Boolean.toString(false));
```

Comment:
What kind of configurable are you expecting here? It's configurable by table's scale currently. Say if the table's properties have these settings(user can use hive-cli to set these properties), it will respect the values from the table.